internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -95,15 +96,21 @@ func unzip(log *logger.Logger, archivePath, dataDir string) (UnpackResult, error
 	}
 
 	hash = metadata.hash[:hashLen]
-
+	var registry map[string][]string
 	if metadata.manifest != nil {
 		pm.mappings = metadata.manifest.Package.PathMappings
 		versionedHome = filepath.FromSlash(pm.Map(metadata.manifest.Package.VersionedHome))
+		registry = metadata.manifest.Package.Flavors
 	} else {
 		// if at this point we didn't load the manifest, set the versioned to the backup value
 		versionedHome = createVersionedHomeFromHash(hash)
 	}
 
+	skipFn, err := skipFnFromZip(log, r, flavor, fileNamePrefix, createVersionedHomeFromHash(hash), registry)
```

Comment:
Shouldn't we use `versionedHome` instead of `createVersionedHomeFromHash(hash)` here?

Reply:
versionedHome at this point can have form of `version-snapshot-hash` as defined in a mapping here https://github.com/elastic/elastic-agent/blob/2985c322f5d383dc1b3b89e4d1c227225873d850/dev-tools/mage/settings.go#L361

this is not what is physical path in a zip file. this is resulting after unpacking